src/ImageSharp/Formats/Tiff/TiffMetadata.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    /// <summary>
+    /// Provides Tiff specific metadata information for the image.
+    /// </summary>
+    public class TiffMetadata : IDeepCloneable
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TiffMetadata"/> class.
+        /// </summary>
+        public TiffMetadata()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TiffMetadata"/> class.
+        /// </summary>
+        /// <param name="other">The metadata to create an instance from.</param>
+        private TiffMetadata(TiffMetadata other)
+        {
+            this.ByteOrder = other.ByteOrder;
+            this.BitsPerPixel = other.BitsPerPixel;
+            this.Compression = other.Compression;
+            this.PhotometricInterpretation = other.PhotometricInterpretation;
+            this.XmpProfile = other.XmpProfile != null ? new byte[other.XmpProfile.Length] : null;
+            other.XmpProfile?.AsSpan().CopyTo(this.XmpProfile.AsSpan());
+        }
+
+        /// <summary>
+        /// Gets the byte order.
+        /// </summary>
+        public ByteOrder ByteOrder { get; internal set; }
```

Comment:
Why are the setters here internal. We want to allow people to change them.

Reply:
Those are internal because they seem only to make sense to retrieve by the user when decoding. When the image is encoded most of those properties are implicitly determined by the Encoding options.

Reply:
Hmmm... that’s different from the way others work. Will review again

Reply:
I notice these properties exist on both the main metadata and individual frames. Is that wise? required?

I.E If they are repeated for each frame then what frame does the parent metadata represent?

Reply:
@JimBobSquarePants each frame has own `ByteOrder`, seems at the moment parent metadata represent byteOrder of root frame, I think this is a discrepancy and here you need to think about how to make it more correct

Reply:
Are you sure? `ByteOrder` is not present in the `TiffFrameMetada` class so I would assume that it is not representing an individual frame at all but represents the global Tiff.

`BitsPerPixel`, `Compression`, and  `PhotometricInterpretation` are present in both classes which appears to be a mistake unless they are global properties that can be overwritten by individual frames. (_In which case I would simply read the value then set as default for each frame before overwriting_)

Reply:
oops, yes, the byte order is set in file header, everything else can be different for frames

Reply:
Ok. So those other properties should be on the frame metadata only.